packages-exp/auth-exp/src/platform_browser/iframe/gapi.test.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { FirebaseError } from '@firebase/util';
+
+import { testAuth } from '../../../test/mock_auth';
+import { Auth } from '../../model/auth';
+import { AUTH_WINDOW } from '../auth_window';
+import { _loadGapi, _resetLoader } from './gapi';
+
+use(sinonChai);
+use(chaiAsPromised);
+
+describe('src/platform_browser/iframe/gapi', () => {
+  let library: unknown;
+  let tag: HTMLElement;
+  let auth: Auth;
+  function onJsLoad(): void {
+    AUTH_WINDOW.gapi = library as typeof gapi;
+    AUTH_WINDOW[(tag as HTMLScriptElement).src.split('onload=')[1]]();
+  }
+
+  beforeEach(async () => {
+    const head = document.createElement('div');
+    tag = document.createElement('script');
+
+    sinon.stub(document, 'createElement').returns(tag);
```

Comment:
this is a lot of stubs... can we just make a util function and stub that out like we do with window.navigator and whatnot

Reply:
Done